### PR TITLE
Add support for hashicorp vault #164

### LINKED ISSUE #164: Add support for Hashicorp Vault
At the moment, all tokens and credentials are stored in the inventory file. This file is plain-text which is a potential security issue if unauthorized people get access to it.
It would be a great option to use Hashicorp Vault to store credentials.
Option to store credentials directly in the inventory file should remain for backwards compatibility

### PR DESCRIPTION
**Pull request by patrikspiess**
Add a new vault helper class for approle login
- Update config class
- Update inventory class
- Add Tests
- Update documentation

closes #164

### CODE REVIEW

File: fotoobo.yaml.sample [hunk 1]
```diff
@@ -58,3 +55,16 @@ logging:
...
+# Instead of storing credentials in the inventory file you may use VAULT as a placeholder. All asset
+# attributes that are VAULT will be retreived from the Hashicorp Vault service specified here.
+# The Hashicorp Vault service will use approle login to get the data.
+vault:
+    url: https://vault.local
+    namespace: vault_namespace
+    data_path: /v1/kv/data/fotoobo
+    role_id: ...
+    secret_id: ...
+    token_file: ~/.cache/token.key
```
Comment (lucmurer) [inline fotoobo.yaml.sample:70]: Possibly add a comment that this setting is optional and may pose a security risk

File: fotoobo/helpers/config.py [hunk 2]
```diff
@@ -71,15 +73,24 @@ def load_configuration(self, config_file: Union[Path, None]) -> None:
...
-
+                self.logging = loaded_config.get("logging", {})
+                self.audit_logging = loaded_config.get("audit_logging", {})
                 self.no_logo = loaded_config.get("no_logo", self.no_logo)
-
                 self.ssl_verify = loaded_config.get("ssl_verify", self.ssl_verify)
 
+                self.vault = loaded_config.get("vault", {})
+                if self.vault:
+                    if (
```
Comment (lucmurer) [inline fotoobo/helpers/config.py:83]: Pls add a comment describing that this is a completeness check for the keys

File: fotoobo/helpers/vault.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        if token_file:
+            self.token_file = Path(token_file).expanduser()
+            log.debug("vault_client_token_file: '%s'", self.token_file)
+            self.load_token()
+
+    def get_data(self, timeout: int = 3) -> Any:
+        """Get data from the key/value store
+
+        Args:
+            path:       The path to your data in the key/value store
```
Comment (lucmurer) [inline fotoobo/helpers/vault.py:75]: Parameter does not exist and is not used?!

File: fotoobo/helpers/vault.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            timeout:    The time before a request to the vault is cancelled
+
+        Returns:
+            The date from the vault
+        """
+        if not self.token:
+            self.get_token()
```
Comment (lucmurer) [inline fotoobo/helpers/vault.py:82]: Response from get_token() is not used -> What happens if no token can be got?

File: fotoobo/inventory/inventory.py [hunk 5]
```diff
@@ -174,6 +191,27 @@ def _load_inventory(self) -> None:
...
+            for attribute in dir(asset):
+                if not attribute.startswith("_"):
+                    if getattr(asset, attribute) == "VAULT":
+                        try:
+                            setattr(asset, attribute, self.vault_data[name][attribute])
+                            log.debug(
+                                "Vault attribute '%s.%s' replaced successfully", name, attribute
+                            )
+                        except KeyError:
+                            log.warning("Vault attribute '%s.%s' not found", name, attribute)
```
Comment (lucmurer) [inline fotoobo/inventory/inventory.py:213]: Perhaps we should open up an issue, that WARNING is the default log level (if nothing set, currently default is no logging if not requested), so users will see this warning.

**Review by lucmurer — CHANGES_REQUESTED** [review]
Good work! Still some small things to polish though.

File: fotoobo/inventory/inventory.py (hunk 5 above)
Comment (lucmurer) [reply fotoobo/inventory/inventory.py:213]: https://github.com/migros/fotoobo/issues/170

File: fotoobo/helpers/config.py (hunk 2 above)
Comment (patrikspiess) [reply fotoobo/helpers/config.py:83]: Adde the following comment:
  
    # Check if all the mandatory settings in vault section are given by substracting
    # the list of keys in vault config from list of mandatory keys.

File: fotoobo.yaml.sample (hunk 1 above)
Comment (patrikspiess) [reply fotoobo.yaml.sample:70]: Added the following info:

    # This section is optional and may pose a security issue. In future versions we may also read the
    # role_id and secret_id from the system environment.

File: fotoobo/helpers/vault.py (hunk 3 above)
Comment (patrikspiess) [reply fotoobo/helpers/vault.py:75]: removed

File: fotoobo/helpers/vault.py (hunk 4 above)
Comment (patrikspiess) [reply fotoobo/helpers/vault.py:82]: Added a check and raise an exception if no token could be retreived

**Review by lucmurer — APPROVED** [review]

**Pull request merged**